### PR TITLE
improve projects and events

### PR DESCRIPTION
**Pull request by linhvth**
This is a pull request for reviewing and merging the changes in Projects and Events to main program.
Please review it, and ask for change if needed.

### CODE REVIEW

**Review by npnkhoi — CHANGES_REQUESTED** [review]
Please push your `package.json` and `package-lock.json`. That file is important for other members to install the same packages as yours.

File: frontend/src/components/Events.js [hunk 1]
```diff
@@ -1,45 +1,70 @@
...
 }
 
-const Projects = () => {
+const Events = () => {
+  const [events, setEvents] = useState([])
+  useEffect(() => {
+    const fetchEvents = async () => {
+      const res = await axios.get(`${BACKEND_URL}/api/events?populate=*`)
+      console.log(`${BACKEND_URL}/api/events?populate=*`);
+      console.log(res.data.data);
```
Comment (npnkhoi) [inline frontend/src/components/Events.js:43]: please delete these

File: frontend/src/components/Projects.js [hunk 2]
```diff
@@ -1,62 +1,67 @@
...
+    <div className="container pt-4 col-9">
+      <SectionHeader text="Projects" />
+      <div className="row my-5">
+        {projects
+        .map(({attributes}) => 
+          <div class="col-md-4 mb-5">
+            <Card 
+              image={ attributes.featureImage } 
+              title={ attributes.name } 
+              date={ moment().startOf('day').fromNow() }
```
Comment (npnkhoi) [inline frontend/src/components/Projects.js:60]: Is this a hack? Looks like we are measuring distance from the start of the date?

File: frontend/src/components/SingleEvent.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  }, {
+    encodeValuesOnly: true,
+  });
+
+  const [event, setEvent] = useState([])
+  useEffect(() => {
+    const fetchEvent = async () => {
+      const res = await axios.get(`${BACKEND_URL}/api/events?${query}`)
+      console.log('The URL of data is ' + `${BACKEND_URL}/api/events?${query}`);
+      console.log(res.data.data);
```
Comment (npnkhoi) [inline frontend/src/components/SingleEvent.js:28]: Delete

File: frontend/src/components/SingleProject.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      console.log(res.data.data);
+      setProject(res.data.data)
+    }
+
+    fetchProject()
+  }, [])
+
+
+  const FeatureImg = (imgsrc) => {
+    console.log(imgsrc.imgsrc.data);
```
Comment (npnkhoi) [inline frontend/src/components/SingleProject.js:38]: Delete

File: frontend/src/components/SingleProject.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  }, {
+    encodeValuesOnly: true,
+  });
+
+  const [project, setProject] = useState([])
+  useEffect(() => {
+    const fetchProject = async () => {
+      const res = await axios.get(`${BACKEND_URL}/api/projects?${query}`)
+      console.log('The URL of data is ' + `${BACKEND_URL}/api/projects?${query}`);
+      console.log(res.data.data);
```
Comment (npnkhoi) [inline frontend/src/components/SingleProject.js:29]: Delete

File: frontend/src/styles/bootstrap/overrides.scss [hunk 6]
```diff
@@ -25,10 +25,16 @@ $position-values: (
   100: 100%
 );
 
-$border-radius: 1.5rem;
+$border-radius: 1.2rem;
 
-.card:hover{box-shadow: 3px 3px 10px -4px #00196E; cursor: pointer;}
-.inner{overflow: hidden;}
+// .card:hover{box-shadow: 3px 3px 10px -4px #00196E; cursor: pointer;}
+// .inner{overflow: hidden;}
```
Comment (npnkhoi) [inline frontend/src/styles/bootstrap/overrides.scss:31]: Is this unused now?

**Review by npnkhoi — APPROVED** [review]
Very good start. Should address the changes and merge first, then address the other features (author name, etc.) later.

File: frontend/src/components/Projects.js (hunk 2 above)
Comment (linhvth) [reply frontend/src/components/Projects.js]: I just updated the code for this part, please review

File: frontend/src/components/SingleEvent.js (hunk 3 above)
Comment (linhvth) [reply frontend/src/components/SingleEvent.js]: done

File: frontend/src/components/SingleProject.js (hunk 4 above)
Comment (linhvth) [reply frontend/src/components/SingleProject.js]: done

File: frontend/src/components/SingleProject.js (hunk 5 above)
Comment (linhvth) [reply frontend/src/components/SingleProject.js]: done

File: frontend/package.json [hunk 7]
```diff
@@ -10,10 +10,14 @@
...
     "bootstrap": "^5.1.3",
     "bootstrap-icons": "^1.8.1",
+    "javascript-time-ago": "^2.5.4",
+    "moment": "^2.29.3",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-markdown": "^8.0.3",
     "react-router-dom": "^6.2.2",
     "react-scripts": "5.0.0",
+    "react-time-ago": "^7.2.1",
```
Comment (npnkhoi) [inline frontend/package.json:20]: Hmm, 2 new packages for time ago sounds a bit overkill. But it seems to work now, so I dont mind much.

**Review by npnkhoi — APPROVED** [review]
Looks ok. Please merge.

**Pull request merged**